### PR TITLE
Add file modification warning to change accessibility combobox in legacy resx editor

### LINKED ISSUE #9034: Form1.Designer.cs file is lost when changing the access modifier from "No code generation" to "Public" for the Form1.resx file
## Visual Studio Version

Version 17.7.0 Preview 2.0 [33718.333.main]

**Have you experienced this same issue with previous previews of VisualStudio?**
Yes, it is not a regression issue, also can repro on VS 16.11.26.

## Steps to Reproduce

1. Create a C# .NET Core WinForms application.
2. Open Form1.resx file.
3. Change the access modifier from "No code generation" to "Public".
4. Observe solution explorer. 

## Expected Behavior
Form1.Designer.cs file exist and generate Form11.Designer.cs file under Form1.resx file.
image

## Actual Behavior
Form1.Designer.cs file is lost.
image

### PR DESCRIPTION
**Pull request by adamint**
Adds the following warning before being allowed to change the class accessibility:
image

This is an auto-generated linked file, and per comment in this folder "We actually don't want users to edit Form RESX file." There is a warning when modifying the values of this file, this editor will eventually be replaced, it's behavior we don't support anyway, and not a trivial fix, so I think adding the warning is enough. The correct behavior for users is to not modify the file at all - doesn't really make sense why it wasn't just made readonly.

Fixes #9034 

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/project-system/pull/9109)

### CODE REVIEW

File: src/Microsoft.VisualStudio.Editors/DesignerFramework/AccessModifierComboBox.vb [hunk 1]
```diff
@@ -81,6 +83,7 @@ Namespace Microsoft.VisualStudio.Editors.DesignerFramework
         ' checked the project system yet)
         ' This field should only be accessed through the CustomToolsRegistered property.
         Private _customToolsRegistered As Boolean?
+        Private Shared _allowEdit As Boolean
```
Comment (drewnoakes) [inline src/Microsoft.VisualStudio.Editors/DesignerFramework/AccessModifierComboBox.vb:86]: This is static right? Does this then break if you have multiple editors open at the same time?

Comment (adamint) [reply src/Microsoft.VisualStudio.Editors/DesignerFramework/AccessModifierComboBox.vb:86]: Good call. The variable needs to be reset afterwards

Comment (drewnoakes) [reply src/Microsoft.VisualStudio.Editors/DesignerFramework/AccessModifierComboBox.vb:86]: If it's static, there'll always be a race condition in the case that multiple editors are opened at once, such when re-opening a solution. We need to find a way to make this non-static.

Comment (tmeschter) [reply src/Microsoft.VisualStudio.Editors/DesignerFramework/AccessModifierComboBox.vb:86]: Any reason we can't make it non-`Shared`?

Comment (adamint) [reply src/Microsoft.VisualStudio.Editors/DesignerFramework/AccessModifierComboBox.vb]: Made it instance. My IDE was yelling at me before? I don't know why.

File: src/Microsoft.VisualStudio.Editors/DesignerFramework/AccessModifierComboBox.vb [hunk 2]
```diff
@@ -438,7 +441,23 @@ Namespace Microsoft.VisualStudio.Editors.DesignerFramework
...
                 If codeGenerator.DisplayName.Equals(value, StringComparison.CurrentCultureIgnoreCase) Then
-                    TrySetCustomToolValue(codeGenerator.CustomToolValue)
+                    Dim Designer = CType(RootDesigner, ResourceEditorRootDesigner)
+                    If Designer IsNot Nothing Then
+                        Dim ResourceView As ResourceEditorView = Designer.GetView()
+                        ' We let the base class handle the read only mode
+                        ' As mentioned in ResourceEditorDesignerLoader, "We actually don't want users to edit Form RESX file"
+                        ' so we just need to add the same warning as there for now until the new resource explorer is released
+                        If ResourceView IsNot Nothing AndAlso Not ResourceView.ReadOnlyMode
+                           If ResourceView.DsMsgBox(My.Resources.Microsoft_VisualStudio_Editors_Designer.RSE_Err_UpdateADependentFile, MessageBoxButtons.YesNo, MessageBoxIcon.Warning, MessageBoxDefaultButton.Button2, HelpIDs.Err_EditFormResx) = DialogResult.Yes Then
```
Comment (drewnoakes) [inline src/Microsoft.VisualStudio.Editors/DesignerFramework/AccessModifierComboBox.vb:451]: I wonder what the significance of `HelpIDs.Err_EditFormResx` is. Does that link to some documentation that'd need to be updated?

**Review by drewnoakes — APPROVED** [review]

**Review by haileymck — APPROVED** [review]

File: src/Microsoft.VisualStudio.Editors/DesignerFramework/AccessModifierComboBox.vb (hunk 2 above)
Comment (adamint) [reply src/Microsoft.VisualStudio.Editors/DesignerFramework/AccessModifierComboBox.vb:451]: It does not - this addresses a gap when editing a resx that shouldn't be edited - the warning that should be displayed was not. So no documentation needs to be updated

**Pull request merged**